subscription/views.py

```diff
@@ -999,41 +1000,73 @@ def post(self, request):
 
 
 ## New APIs for individual 
-# def is_paid_page(uri):
-#     patterns = UriPattern.objects.values_list('pattern', flat=True)
-#     return any(re.match(pattern, uri) for pattern in patterns)
+# def is_premium_page(url):
+#     premium_urls = PremiumUsageUnits.objects.values_list('url', flat=True)
+#     return any(re.match(premium_url, url) for premium_url in premium_urls)
+
+# def get_premium_units(url):
+#     logger.debug("get_premium_units "+url)
+#     try:
+#         premium_page = PremiumUsageUnits.objects.get(url=url)
+#         return premium_page.units_consumed
+#     except PremiumUsageUnits.DoesNotExist:
+#         # If no exact match, try regex matching
+#         for premium_page in PremiumUsageUnits.objects.all():
+#             if re.match(premium_page.url, url):
+#                 return premium_page.units_consumed
+#     return 1  # Default to 1 unit if not found in PremiumUsageUnits
+
+def get_premium_units(url):
+    logger.debug("get_premium_units " + url)
+    try:
+        # First, try an exact match
+        premium_page = PremiumUsageUnits.objects.get(url=url)
+        return premium_page.units_consumed
+    except PremiumUsageUnits.DoesNotExist:
+        # If no exact match, try containment check
+        for premium_page in PremiumUsageUnits.objects.all():
+            if premium_page.url in url:
+                return premium_page.units_consumed
+        
+        # If still no match, try regex matching as a fallback
+        for premium_page in PremiumUsageUnits.objects.all():
+            if re.match(premium_page.url, url):
+                return premium_page.units_consumed
+    
+    return 1  # Default to 1 unit if not found in PremiumUsageUnits
 
 class CheckLimit(APIView):
     requireApiKey = False
 
     def get(self, request):
-        # timestamp = request.GET.get('timestamp')
-        # complete_uri = request.GET.get('complete_uri')
+        complete_uri = request.GET.get('uri')
         party_id = request.GET.get('party_id')
-        # logger.debug("CheckLimit view called")
-
-        if not all([party_id]):
-            return Response({"error": "Missing party_id parameter"}, status=status.HTTP_400_BAD_REQUEST)
 
-        # logger.debug("all variables present")
-        # Check if the requested URI is a paid page
-        # if not is_paid_page(complete_uri):
-        #     return Response({"status": "ok", "message": "Free page access granted"})
-
-        # logger.debug("paid page")
+        if not all([party_id, complete_uri]):
+            return Response({"error": "Missing required parameters"}, status=status.HTTP_400_BAD_REQUEST)
 
+        units_required = get_premium_units(complete_uri)
+        
+        try:
+            limit_value_object = LimitValue.objects.get(limitId=1)
```

Comment:
This is incorrect. No constant expected. Should get by partnerId. We verified that the partnerId is provided in the proxy invocation.